src/static/js/modules/youtube.js

```diff
@@ -0,0 +1,181 @@
+   Youtube Initialization
+   Used on at least `/the-bureau/`.
+   ========================================================================== */
+
+'use strict';
+
+var $ = require( 'jquery' );
+var jsLoader = require( './util/js-loader' );
+var viewportDimensions = require( './breakpoint-handler' ).getViewportDimensions;
+
+function init() {
+  return $( '.youtube-player_container' ).createYouTube();
+}
+
+// YouTube constructor.
+// @param element {Element} DOM Element.
+// @param options {Object} Object used to customize YouTube.
+// @returns {Object} An YouTube instance.
+function YouTube( element, options ) {
+  var playerOptions = this.defaults.playerOptions;
+  this.$element = $( element );
+  $.extend( true, playerOptions, options );
+  this.init();
+}
+
+// YouTube methods and shared properties
+YouTube.prototype = {
+
+  constructor: YouTube,
+
+  defaults: {
+    playerOptions: {
+      height:     '410',
+      width:      '640',
+      videoId:    '',
+      playerVars: {
+        autoplay:         1,
+        suggestedQuality: 'highres'
+      },
+      events: {
+        onReady:       $.noop,
+        onStateChange: $.noop
+      }
+    },
+    playerInitialized: false
+  },
+
+  _bindAll: function() {
+    var self = this;
+    var proxy = $.proxy;
+
+    // TODO: Swap this out for Function.bind
+    // and Object.keys.
+    $.each( this, function( key, _function ) {
+      if ( $.isFunction( _function ) ) {
+        self[key] = proxy( _function, self );
+      }
+    } );
+
+    return this;
+  },
+
+  _embedScript: function() {
+    window.onYouTubeIframeAPIReady = this.initPlayer;
+    jsLoader.loadScript( 'http://www.youtube.com/iframe_api' );
+
+    return this;
+  },
+
+  init: function() {
+    this._bindAll();
+    this.initUI();
+
+    var YouTubeEvents = this.defaults.playerOptions.events;
+    YouTubeEvents.onReady = this.onPlayerReady;
+    YouTubeEvents.onStateChange = this.onPlayerStateChange;
+
+    return this;
+  },
+
+  initUI: function() {
+    var ui = this.$element;
+    ui.on( 'click', '.youtube-player_play-btn', this.play );
+    ui.on( 'click', '.youtube-player_close-btn', this.close );
+
+    return this;
+  },
+
+  destroy: function() {
+    var self = this;
+    this.$element.remove();
+
+    $.each( this, function( key ) {
+      delete self[key];
+    }, this );
+
+    return this;
+  },
+
+  initPlayer: function() {
+    var YT = window.YT;
+    if ( YT && YT.Player ) {
+      new YT.Player( 'youtube-player', this.defaults.playerOptions );
+      this.playerInitialized = true;
+    } else {
+      this._embedScript();
+    }
+
+    return this;
+  },
+
+  onPlayerStateChange: function( event ) {
+    if ( event.data === window.YT.PlayerState.ENDED ) {
+      this.$element.removeClass( 'yt-playing' );
+    }
+
+    return this;
+  },
+
+  onPlayerReady: function( params ) {
+    if ( params && params.target ) {
+      this.player = params.target;
+    }
+  },
+
+  play: function( e ) {
```

Comment:
I've used `evt` in the past because `event` gets highlighted by not-so-smart syntax highlighters... but either is fine... consistency is key for making searches easy for refactoring.